### PR TITLE
Add search by properties and render property view

### PR DESCRIPTION
**Pull request by benbaumann95**
### Description of change

Quite a long list of things... Probably best to go through commits..
**But in summary:**
- Adds axios to talk to our API
- Adds necessary react libraries e.g react-router-dom
- Adds front-end libraries 
- Adds jest unit testing
- Adds .env to use environment variables
- Add components to render Properties table and view pages

### Story Link

Part of:
https://trello.com/c/V1OfUzOV/98-as-a-user-i-need-to-search-for-properties-using-the-address-or-postcode-so-that-i-can-see-a-list-of-properties
Part of:
https://trello.com/c/nS0rF2iL/153-as-a-user-i-need-to-see-the-address-postcode-and-property-type-so-that-i-know-what-property-im-looking-at

### Known issues [OPTIONAL]

- Probably need to revaluate the use of Rails in this application.. It seems it may not be needed but we will know more after deciding how to do authentication

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,6 +29,11 @@ jobs:
       run: |
         sudo apt-get -yqq install libpq-dev
 
+    - name: Install Yarn
```
Comment (benbaumann95) [inline .github/workflows/ci.yml:32]: Chore to add caching to this at a later stage..

File: app/javascript/stylesheets/components/button.scss [hunk 2]
```diff
@@ -0,0 +1,29 @@
+.govuk-button {
```
Comment (benbaumann95) [inline app/javascript/stylesheets/components/button.scss:1]: Should discuss a better way of doing this I think. Quick and dirty for now..
Don't like using `!important` to override but it is needed to override the govuk style

**Review by LBHABEBIC — APPROVED** [review]
Hey looks good
Not sure if you've had a look at https://github.com/LBHackney-IT/lbh-social-care there is a lot of components that can be reused including a Spinner for loading.

**Comment by benbaumann95** [comment]
> Hey looks good
> Not sure if you've had a look at https://github.com/LBHackney-IT/lbh-social-care there is a lot of components that can be reused including a Spinner for loading.

Hey thanks a lot for taking a look!
First time doing React so appreciate it, took some of the open source react projects on here for inspiration.
Will take a look, thanks!

File: app/javascript/src/utils/api/properties.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  return data
+}
+
+export const getProperty = async (propertyReference) => {
+  const { data } = await axios.get(
+    `http://localhost:3000/api/v1/properties/${propertyReference}`
+  )
+
+  return data
+}
```
Comment (EGiataganas) [inline app/javascript/src/utils/api/properties.js:19]: How about moving the `server url` into a constant variable and add a dotenv? something like that

```JS
const API = 'http://localhost:3000/'
```

```JS
API + `api/v1/properties/?q=${params}`
```

**Review by neilvanbeinum — APPROVED** [review]

**Pull request merged**